clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/ChainResolver.java

```diff
@@ -72,62 +76,67 @@ public ChainResolver(final OperationsCodec<K, V> codec, Expiry<? super K, ? supe
-              duration = expiry.getExpiryForCreation(key, result.getValue());
-              if (duration == null) {
-                result = null;
-                continue;
-              }
-            } else {
-              duration = expiry.getExpiryForUpdate(key, new ValueSupplier<V>() {
-                @Override
-                public V value() {
-                  return previousResult.getValue();
+        if (expiry != Expirations.noExpiration()) {
+          if(operation.isExpiryAvailable()) {
+            expirationTime = operation.expirationTime();
+            if (expirationTime == Long.MIN_VALUE) {
+              continue;
+            }
+            if (now >= expirationTime) {
+              result = null;
+            }
+          } else {
+            Duration duration;
+            try {
+              if(previousResult == null) {
+                duration = expiry.getExpiryForCreation(key, result.getValue());
+                if (duration == null) {
+                  result = null;
+                  continue;
+                }
+              } else {
+                duration = expiry.getExpiryForUpdate(key, new ValueSupplier<V>() {
+                  @Override
+                  public V value() {
+                    return previousResult.getValue();
+                  }
+                }, result.getValue());
+                if (duration == null) {
+                  continue;
                 }
-              }, result.getValue());
-              if (duration == null) {
-                continue;
               }
+            } catch (Exception ex) {
+              LOG.error("Expiry computation caused an exception - Expiry duration will be 0 ", ex);
+              duration = Duration.ZERO;
+            }
+            compacted = true;
+            if(duration.isInfinite()) {
+              expirationTime = Long.MIN_VALUE;
+              continue;
+            }
+            long time = TIME_UNIT.convert(duration.getLength(), duration.getTimeUnit());
+            expirationTime = time + operation.timeStamp();
+            if(now >= expirationTime) {
+              result = null;
             }
-          } catch (Exception ex) {
-            LOG.error("Expiry computation caused an exception - Expiry duration will be 0 ", ex);
-            duration = Duration.ZERO;
-          }
-          compacted = true;
-          if(duration.isInfinite()) {
-            expirationTime = Long.MIN_VALUE;
-            continue;
-          }
-          long time = TIME_UNIT.convert(duration.getLength(), duration.getTimeUnit());
-          expirationTime = time + operation.timeStamp();
-          if(now >= expirationTime) {
-            result = null;
           }
         }
       } else {
         payload.flip();
         chainBuilder = chainBuilder.add(payload);
       }
     }
-    if(result == null) {
-      compacted = keyMatch > 0;
+
+    compacted = (result == null) ? (keyMatch > 0) : (compacted || keyMatch > 1);
```

Comment:
There was no particular reason behind that. There are too many if-else blocks already in that method. Including the ones following this line. Since this was the simplest if-else block that could easily be replaced with a ternary operator I went for it. There is no other reason. I can revert it, if is affecting readability.